EIPS/eip-draft_rules_engine.md

```diff
@@ -0,0 +1,186 @@
+
+    /**
+        @notice Adds a new RuleSet onto the hierarchy of a RuleTree.
+        @dev RuleSets can have child RuleSets, but they will only be called if the parent's Rules execute to create boolean 'true'.
+        @param _owner           Owner/ID of the RuleTree
+        @param _ruleSetName     ID/Name of the RuleSet
+        @param _desc            Verbose description of the RuleSet
+        @param _parentRSName    ID/Name of the parent RuleSet, to which this will be added as a child
+        @param _severalFailFlag Indicator of whether or not the RuleSet's execution (as failure) will result in a failure of the RuleTree.  (This flag only applies to leaves in the RuleTree.)
+        @param _useAndOp        Indicator of whether or not the rules in the RuleSet will execute with 'AND' between them.  (Otherwise, it will be 'OR'.)
+        @param _failQuickFlag   Indicator of whether or not the RuleSet's execution (as failure) should immediately stop the RuleTree.
+    */    
+    function addRuleSet(address _owner, bytes32 _ruleSetName, string calldata _desc, bytes32 _parentRSName, bool _severalFailFlag, bool _useAndOp, bool _failQuickFlag) external;
+
+    /**
+        @notice Adds a new Rule into a RuleSet.
+        @dev Rule types can be implemented as any type of action (greater than, less than, etc.)
+        @param _owner           Owner/ID of the RuleTree
+        @param _ruleSetName     ID/Name of the RuleSet to which the Rule will be added
+        @param _ruleName        ID/Name of the Rule being added
+        @param _attrName        ID/Name of the Attribute upon which the Rule is invoked
+        @param _ruleType        ID of the type of Rule
+        @param _rightHandValue  The registered value to be used by the Rule when performing its action upon the Attribute
+        @param _notFlag         Indicator of whether or not the NOT operator should be performed on this Rule.
+    */    
+    function addRule(address _owner, bytes32 _ruleSetName, bytes32 _ruleName, bytes32 _attrName, uint _ruleType, string calldata _rightHandValue, bool _notFlag) external;
+
+    /**
+        @notice Executes a RuleTree.
+        @param _owner           Owner/ID of the RuleTree
+    */
+    function executeRuleTree(address _owner) external returns (bool);
+    
+    /**
+        @notice Removes a RuleTree.
+        @param _owner           Owner/ID of the RuleTree
+    */
+    function removeRuleTree(address _owner) external returns (bool);    
+}
+```
+
+### Considerations
+
+An argument could be made for interface functions that allow a RuleTree's owner to include others users as executors of the RuleTree.
+
+Another argument could be made for interface functions that allow an administrator to configure the origin point of an Attribute, such as whether the Attribute's value comes from a data structure (internal to the rules engine contract) or from calling a contract method (like an implementation of the Diamond Standard).
```

Comment:
It would be helpful to link to the Diamond Standard EIP here in addition to the references section.

Reply:
Thanks for the help, @lightclient!